docker/docker.ts

```diff
@@ -211,7 +211,7 @@ async function buildAndPushImageWorkerAsync(
 
         for (const stage of stages) {
             await tagAndPushImageAsync(
-                localStageImageName(imageName, stage), repositoryUrl, tag, /*imageId:*/ undefined, logResource);
+                localStageImageName(imageName, stage), repositoryUrl, stage, /*imageId:*/ undefined, logResource);
```

Comment:
was passing along wrong variable, leading to the cache-from builds not being pushed to the same location we were pulling from.  Validated that fix works for fixing cloud tests over here: https://github.com/pulumi/pulumi-cloud/pull/639